cypress/support/ppmCustomerShared.js

```diff
@@ -55,17 +56,82 @@ export function fillOutAboutPage(selectAdvance) {
     expect(loc.pathname).to.match(/^\/moves\/[^/]+\/shipments\/[^/]+\/weight-tickets/);
   });
 }
 
+export function signInAndNavigateToWeightTicketPage(userId) {
+  cy.apiSignInAsUser(userId);
+  cy.wait('@getShipment');
+  cy.screenshot();
+  cy.get('button[data-testid="button"]').contains('Upload PPM Documents').click();
+  fillOutAboutPage(true);
+}
+
+export function submitWeightTicketPage(options) {
+  fillOutWeightTicketPage(options);
+  navigateFromWeightTicketPage();
+}
+
+export function fillOutWeightTicketPage(options) {
+  cy.get('input[name="vehicleDescription"]').clear().type('Kia Forte').blur();
+
+  if (options?.useConstructedWeight) {
+    cy.get('input[name="emptyWeight"]').clear().type('1000').blur();
+    cy.get('input[name="missingEmptyWeightTicket"]').check({ force: true });
+    cy.intercept('/internal/uploads').as('uploadFile');
+    cy.upload_file('.emptyDocument.filepond--root', 'constructedWeight.xls');
+    cy.get('[data-filepond-item-state="processing-complete"]', { timeout: fileUploadTimeout }).should('have.length', 1);
+    cy.get('input[name="fullWeight"]').clear().type('3000');
+    cy.get('input[name="missingFullWeightTicket"]').check({ force: true });
+    cy.intercept('/internal/uploads').as('uploadFile');
+    cy.upload_file('.fullDocument.filepond--root', 'constructedWeight.xls');
+    cy.get('[data-filepond-item-state="processing-complete"]', { timeout: fileUploadTimeout }).should('have.length', 1);
+  } else {
+    cy.get('input[name="emptyWeight"]').clear().type('1000').blur();
+    cy.intercept('/internal/uploads').as('uploadFile');
+    cy.upload_file('.emptyDocument.filepond--root', 'sampleWeightTicket.jpg');
+    cy.get('[data-filepond-item-state="processing-complete"]', { timeout: fileUploadTimeout }).should('have.length', 1);
+    cy.get('input[name="fullWeight"]').clear().type('3000');
+    cy.intercept('/internal/uploads').as('uploadFile');
+    cy.upload_file('.fullDocument.filepond--root', 'sampleWeightTicket.jpg');
+    cy.get('[data-filepond-item-state="processing-complete"]', { timeout: fileUploadTimeout }).should('have.length', 1);
+  }
+
+  cy.get('.tripWeightTotal').contains('Trip weight: 2,000 lbs');
+
+  if (options?.hasTrailer) {
+    cy.get('input[name="ownsTrailer"][value="true"]').check({ force: true });
+    if (options?.ownTrailer) {
+      cy.get('input[name="trailerMeetsCriteria"][value="true"]').check({ force: true });
+      cy.intercept('/internal/uploads').as('uploadFile');
+      cy.upload_file('.proofOfTrailerOwnershipDocument.filepond--root', 'trailerOwnership.pdf');
+      cy.get('[data-filepond-item-state="processing-complete"]', { timeout: fileUploadTimeout }).should(
+        'have.length',
+        1,
+      );
+    } else {
+      cy.get('input[name="trailerMeetsCriteria"][value="false"]').check({ force: true });
+    }
+  }
+}
+
+export function navigateFromWeightTicketPage() {
+  cy.get('button').contains('Save & Continue').click();
```

Comment:
I wonder/think the current errors might be resolved if we do this
```suggestion
  cy.get('button').contains('Save & Continue').should('be.enabled').click();
```

Reply:
Okay I'll try this. I was looking at the mp4 on circle and looks like the upload isn't finished before cypress tries to click this button, I wonder if we do need a wait for upload.

Reply:
Yeah now it's failing `expected '<button.usa-button.PPM_saveButton__n3eZE>' to be 'enabled'`

I think I need to add the wait back since it was passing before I took that out